### DISCUSSION TITLE
Blazor Render Mode Interative Auto

### DISCUSSION THREAD

**Discussion by eduardoxvii** [balta-io/forum#524] [Tira Dúvidas] [answered]
Tenho o seguinte código no meu projeto Client:
```
@page "/counter"
@rendermode InteractiveAuto

<PageTitle>Counter</PageTitle>

<h1>Counter</h1>

<p role="status">Current count: @currentCount</p>

<button class="btn btn-primary" @onclick="IncrementCount">Click me1</button>
@code {
    private int currentCount = 0;
    private bool abc = false;

    private void IncrementCount()
    {
        currentCount++;
    }
}
```
No código, a função "IncrementCount" com "@rendermode InteractiveAuto" está muito lento no primeiro acesso, não possui a mesma velocidade do "@rendermode InteractiveServer". 
Nos acessos subsequentes o incremento é automatico, muito rápido! 
Eu pensei que o "@rendermode InteractiveAuto" fosse abrir o websocket na mesma velocidade que o "@rendermode InteractiveServer". 
Está certo isso o primeiro acesso a uma página  ao servidor pelo "@rendermode InteractiveAuto" ser muito mais lento que o "@rendermode InteractiveServer"?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @eduardoxvii, como vai?

Acredito que o primeiro acesso a aplicação como um todo está lento, correto?
Se for isto, é um processo normal, chamado de "warm up", onde o primeiro acesso, do primeiro usuário, dá início a vários processos internos de otimização da aplicação.

Se o servidor onde estiver hospedando der suporte, você pode já fazer uma requisição a determinada página, assim que seu app subir.

Me dá mais detalhes sobre seu ambiente e hospedagem por favor.
